### ISSUE TITLE
Media Source: poor & unclear docs

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/home-assistant.github.io#28940]
### Feedback

I am using HA in container and unable to setup anything related to "Media Source" by using these docs which are supposed to cover all related issues.

1. The "my_address:8123/media-browser/browser" page in HA is not described in Docs.
The page is unclear, has no hints/help/descriptions - so I have to rely on Docs - which do not have any description for this page.

2. In Docs it is specified:
> For Home Assistant OS, Supervised and Container users, this folder is by default configured in the path /media.
> Users of Home Assistant Container can mount a volume of their choice to /media.

but it is no described what this "source" is (available in HA by default w/o any "mounting" etc by a user):
![изображение](https://github.com/home-assistant/home-assistant.io/assets/71872483/5c7d1700-28f2-40ca-9755-eb49ab435a58)
And this "source" has "cdrom", "floppy" & "usb" which are empty. No info is provided how to use them.

3. When describing a `media_dirs` option:
![изображение](https://github.com/home-assistant/home-assistant.io/assets/71872483/1a2b0017-e850-46ba-826a-4f662e838294)
it is not specified whether a user may define any names (not only "local", "recordings") or just a name from some fixed set.

4. When describing a mounting network storage:
> If you want to use media from a network storage, the network storage must first be connected first. Refer to [these instructions on how to connect network storage](https://www.home-assistant.io/common-tasks/os/#network-storage).
> The media from the network storage is then automatically added to the local media browser.

there are no instructions for "HA in container" case - the proposed link seems to be useful for HA OS only.

In short, the Docs has _some_ info but it is rather difficult to get smth from it even for an experienced user (not to mention a beginner).

### URL

https://www.home-assistant.io/integrations/media_source/

### Version

2023.9.2

**Comment by ildar170975** [comment]
up

**Comment by masman69** [comment]
I also have to up-vote.  Spent a fair amount of time getting an Android notification to work with image, and only bc I found examples of that.  Have tried every conceivable way of trying to send a FireTv notification using media source and have had no luck.

The switch from just using the www path is significant.  This needs more docs.  In general all HA documentation needs more examples of the default/standard use cases.

**Comment by dbolton** [comment]
### Document reference and corresponding actual file
I would benefit from a documented example that shows how a file exists on in the actual folder plus how you refer to it. 

For example. the last example in the documentation uses the reference:  
media_content_id: "media-source://media_source/local/videos/favourites/Epic Sax Guy 10 Hours.mp4"

Where would this file exist in the actual folder structure? Is it "media/videos/favorites/Epic Sax Guy 10 Hours.mp4" or somewhere else?

### Document "local"
The word "local" seems to have special meaning for this integration. Could the docs elaborate? 

### My specific example with Nest camera and Gallery Card
I have snapshot motion/person detection camera images saved in "/config/nest/event_media/{{device_id}}". I can view the images in the companion app via Media > Next > Front Door: Recent Events, but can't see them when I add them to a [custom Gallery Card ](https://github.com/TarheelGrad1998/gallery-card) as "- media-source://media_source/local/config/nest/event_media/6a9f9a6fbd8692dc5ef63004c98a09fb" or several other variations (see screenshot below). 
![image](https://github.com/home-assistant/home-assistant.io/assets/712270/aa22b829-824a-4261-b9ea-2ea89b7dc0ae)

### media_dirs integration not found
The docs suggest adding the following to the home assistant configuration.yaml. 

```
homeassistant:
  media_dirs:
    local: /media
```
When I do, I get the warning: "Configuration warnings: Integration error: media_dirs - Integration 'media_dirs' not found". 

Are the docs mistaken or am I misunderstanding something?

**Comment by kepler** [comment]
Bumping. I've been fiddling through pages and two/three years old threads with no luck. I can't get my local mp3 files to play on a soundbar using the media folder. If I place them under `www` they work with `media_content_id: /local/song.mp3`.

There's also this [documentation page](https://www.home-assistant.io/more-info/local-media/setup-media/), but it doesn't actually help. I still get an ERROR when using `media_dirs` and `media-source://media_source/media/song.mp3` (or `media-source://media_source/local/song.mp3` if I change the `media_dirs` to `local: /media` (also tried with `/config/media` and `media`)):
```
Error during call async_play_media: UpnpActionResponseError('Error during async_call(), action: SetAVTransportURI, status: 500, upnp error: 716 (Resource not found)')
```

Funny thing is that if I use an invalid top-level dir like `media-source://media_source/wrongname/song.mp3`, it complains.

**Comment by alextiekam** [comment]
Hi,

i dont know if following helps you, because I do not exactly understand this behaviour:

Snippet:
  - action: camera.snapshot
    target:
      entity_id: camera.eingang
    data:
      filename: /media/eingang.jpg
  - action: notify.notify
    data:
      message: Test
      data:
        image: /media/local/eingang.jpg

I need to save the file in my case to /media/filename.jpg. 
But to call this file within a notification I need to change the path and use /media/local/filename.jpg

Maybe this could help.

**Comment by dbolton** [comment]
Bump
